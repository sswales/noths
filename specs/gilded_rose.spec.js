const { Shop, Item } = require('../src/gilded_rose');

const items = [
  new Item('+5 Dexterity Vest', 10, 20),
  new Item('Aged Brie', 2, 0),
  new Item('Elixir of the Mongoose', 5, 7),
  new Item('Sulfuras, Hand of Ragnaros', 0, 80),
  new Item('Sulfuras, Hand of Ragnaros', -1, 80),
  new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20),
  new Item('Backstage passes to a TAFKAL80ETC concert', 10, 49),
  new Item('Backstage passes to a TAFKAL80ETC concert', 5, 49),
  new Item('Backstage passes to a TAFKAL80ETC concert', 10, 50),
  new Item('Backstage passes to a TAFKAL80ETC concert', 5, 50),
  // This Conjured item does not work properly yet
  new Item('Conjured Mana Cake', 3, 6),
];

describe('Gilded Rose', () => {
  const days = 15;
  const gildedRose = new Shop(items);
  for (let day = 0; day < days; day++) {
    it(`should return expected quality after ${day} day's`, () => {
      expect(gildedRose.updateQuality()).toMatchSnapshot();
    });
  }
});
git s