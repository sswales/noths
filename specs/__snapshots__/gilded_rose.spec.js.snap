// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose should return expected quality after 0 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 19,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 1,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 6,
    "sellIn": 4,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 21,
    "sellIn": 14,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 5,
    "sellIn": 2,
  },
]
`;

exports[`Gilded Rose should return expected quality after 1 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 18,
    "sellIn": 8,
  },
  Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": 0,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 5,
    "sellIn": 3,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 22,
    "sellIn": 13,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 4,
    "sellIn": 1,
  },
]
`;

exports[`Gilded Rose should return expected quality after 2 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 17,
    "sellIn": 7,
  },
  Item {
    "name": "Aged Brie",
    "quality": 4,
    "sellIn": -1,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 4,
    "sellIn": 2,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 23,
    "sellIn": 12,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 3,
    "sellIn": 0,
  },
]
`;

exports[`Gilded Rose should return expected quality after 3 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 16,
    "sellIn": 6,
  },
  Item {
    "name": "Aged Brie",
    "quality": 6,
    "sellIn": -2,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 3,
    "sellIn": 1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 24,
    "sellIn": 11,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 1,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose should return expected quality after 4 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 15,
    "sellIn": 5,
  },
  Item {
    "name": "Aged Brie",
    "quality": 8,
    "sellIn": -3,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 2,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 25,
    "sellIn": 10,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -2,
  },
]
`;

exports[`Gilded Rose should return expected quality after 5 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 14,
    "sellIn": 4,
  },
  Item {
    "name": "Aged Brie",
    "quality": 10,
    "sellIn": -4,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 27,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -3,
  },
]
`;

exports[`Gilded Rose should return expected quality after 6 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 13,
    "sellIn": 3,
  },
  Item {
    "name": "Aged Brie",
    "quality": 12,
    "sellIn": -5,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 29,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -4,
  },
]
`;

exports[`Gilded Rose should return expected quality after 7 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 12,
    "sellIn": 2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 14,
    "sellIn": -6,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 31,
    "sellIn": 7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -5,
  },
]
`;

exports[`Gilded Rose should return expected quality after 8 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 11,
    "sellIn": 1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 16,
    "sellIn": -7,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 33,
    "sellIn": 6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -6,
  },
]
`;

exports[`Gilded Rose should return expected quality after 9 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 10,
    "sellIn": 0,
  },
  Item {
    "name": "Aged Brie",
    "quality": 18,
    "sellIn": -8,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 35,
    "sellIn": 5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -7,
  },
]
`;

exports[`Gilded Rose should return expected quality after 10 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 8,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 20,
    "sellIn": -9,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 38,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -8,
  },
]
`;

exports[`Gilded Rose should return expected quality after 11 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 6,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 22,
    "sellIn": -10,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 41,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -9,
  },
]
`;

exports[`Gilded Rose should return expected quality after 12 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 4,
    "sellIn": -3,
  },
  Item {
    "name": "Aged Brie",
    "quality": 24,
    "sellIn": -11,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 44,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -10,
  },
]
`;

exports[`Gilded Rose should return expected quality after 13 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 2,
    "sellIn": -4,
  },
  Item {
    "name": "Aged Brie",
    "quality": 26,
    "sellIn": -12,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 47,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -11,
  },
]
`;

exports[`Gilded Rose should return expected quality after 14 day's 1`] = `
Array [
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Aged Brie",
    "quality": 28,
    "sellIn": -13,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -12,
  },
]
`;
